h2o-algos/src/main/java/hex/hglm/ComputationStateHGLM.java

```diff
@@ -0,0 +1,152 @@
+    _level2UnitNames = engTask._level2UnitNames;
+    _randomCoefficientNames = engTask._randomCoeffNames;
+    _level2UnitIndex = engTask._level2UnitIndex;
+    initComputationStateHGLM(engTask);
+    _numFixedCoeffs = _beta.length;
+    _numRandomCoeffs = _ubeta[0].length;
+    _numLevel2Unit = _ubeta.length;
+    _nobs = engTask._nobs;
+  }
+
+  /**
+   * set initial values for:
+   * 1. initial fixed coefficients from user or assigned by us;
+   * 2. initial random coefficients from user or randomly assigned;
+   * 3. sigma square;
+   * 4. T matrix value
+   */
+  void initComputationStateHGLM(HGLMTask.ComputationEngineTask engineTask) {
+    int numRandomCoeff = _randomCoefficientNames.length;
+    int numFixCoeff = _fixedCofficientNames.length;
+    // need to initialize the coefficients, fixed and random
+    if (_parms._seed == -1) // set the seed if not set by user
+      _parms._seed = new Random().nextLong();
+    Log.info("Random seed: "+_parms._seed);
+
+    Random random = new Random(_parms._seed);
+    if (_parms._tau_e_var_init > 0.0)
+      _tauEVarE10 = _parms._tau_e_var_init;
+    else
+      _tauEVarE10 = Math.abs(random.nextGaussian());
+
+    _T = new double[numRandomCoeff][numRandomCoeff];
+    if (_parms._initial_t_matrix != null) {
+      grabInitValuesFromFrame(_parms._initial_t_matrix, _T);
+      double[][] transposeT = ArrayUtils.transpose(_T);
+      if (!equal2DArrays(_T, transposeT, 1e-6))
+        throw new IllegalArgumentException("initia_t_matrix must be symmetric but is not!");
+      // make sure matrix is semi positive definite
+      Matrix tMat = new Matrix(_T);
+      if (!tMat.chol().isSPD())
+        throw new IllegalArgumentException("initial_t_matrix must be positive semi definite but is not!");
+    } else {
+      if (_parms._tau_u_var_init > 0.0) {
+        _tauEVarE10 = _parms._tau_u_var_init;
+      } else {
+        _tauEVarE10 = Math.abs(random.nextGaussian());
+      }
+      setDiagValues(_T, _tauEVarE10);
+    }
+    
+    _ubeta = new double[engineTask._numLevel2Units][engineTask._numRandomCoeffs];
+    if ( null != _parms._initial_random_effects) {  // read in initial random values
+      grabInitValuesFromFrame(_parms._initial_random_effects, _ubeta);
+    } else {  // randomly generating random initial values
+      gaussianVector(random, _ubeta, _level2UnitNames.length, numRandomCoeff);
+      ArrayUtils.mult(_ubeta, Math.sqrt(_T[0][0]));
+    }
+    // copy over initial fixed coefficient values
+    if (null != _parms._initial_fixed_effects) {
+      if (_parms._initial_fixed_effects.length != numFixCoeff)
+        throw new IllegalArgumentException("initial_fixed_effects must be an double[] array of size "+numFixCoeff);
+
+      _beta = _parms._initial_fixed_effects;
+    } else {
+      _beta = new double[numFixCoeff];
+      _beta[_beta.length-1] = _parms.train().vec(_parms._response_column).mean();
+    }
+  }
+  
+  public double[] get_beta() { return _beta; }
```

Comment:
I suggest using a Camel case for every getter here in this class.
 
```suggestion
  public double[] getBeta() { return _beta; }
```

Reply:
I changed all of them.